model-api-gen/src/main/kotlin/org/modelix/metamodel/generator/MetaModelGenerator.kt

```diff
@@ -647,6 +647,15 @@ class MetaModelGenerator(val outputDir: Path, val nameConfig: NameConfig = NameC
             for (extended in concept.getDirectSuperConcepts()) {
                 addSuperinterface(extended.conceptWrapperInterfaceClass().parameterizedBy(nodeT))
             }
+
+            for (metaPropertyName in concept.language.languageSet.getMetaPropertyRoots(concept.fqName())) {
```

Comment:
I guess this has to be replaced with `if (concept.getDirectSuperConcepts().isEmpty())` after changing `conceptFqName2RootMetaProperties`.

Reply:
I had a similar approach earlier but in the case of MPS languages this can lead to problems, since interface concepts like `ScopeProvider` don't have any super concepts either. So the `val alias: String get() = null` would be generated into `C_BaseConcept` and `C_ScopeProvider` (among other interfaces). If a concept then both extends these interfaces the Kotlin Compiler is not able to disambiguate these functions even though their implementation is the same.

That's why I went for the approach with the map.

Reply:
Here are some solutions.

### A. Repeat the default implementation in the interface where the conflict happens.

```kotlin
interface ConceptA {
    val alias: String? get() = null
}
interface ConceptB {
    val alias: String? get() = null
}
interface ConceptC : ConceptA, ConceptB {
    override val alias: String? get() = null
}
class ConceptCImpl : ConceptC {
}
```

### B. Don't generate a default implementation and let all concept implement it.

```kotlin
interface ConceptA {
    val alias: String?
}
interface ConceptB {
    val alias: String?
}
interface ConceptC : ConceptA, ConceptB {
}
class ConceptCImpl : ConceptC {
    override val alias: String? get() = null
}
```
### C. Generate a common interface for all concept

```kotlin
interface IConceptProperties {
    val alias: String? get() = null
}
interface ConceptA : IConceptProperties {
}
interface ConceptB : IConceptProperties {
}
interface ConceptC : ConceptA, ConceptB {
}
class ConceptCImpl : ConceptC {
    override val alias: String get() = "abc"
}
```

### D. Generate our own BaseConcept

Similar to option C we could generate a whole new concept that is the super concept of all other concept.
This guarantees that there is a single root in the concept hierarchy and might be useful in the future for solving similar problems.

When choosing this option the concept can just be added during the preprocessing of the languages and doesn't require any changes to the generator itself.

Both option C and D require to generate a unique name for the new interface/concept to avoid duplicate classes/interfaces on the classpath.

Reply:
I went with suggestion C. In order to have a unique name for the generated interface, I let the user configure the name of the interface similar to `registrationHelperName`. This way name clashes can be avoided, even if generated APIs from multiple language sets are used.
If no name is specified, no concept meta-properties will be generated.